ethers-solc/src/resolver.rs

```
@@ -0,0 +1,518 @@
+        let mut edges = Vec::with_capacity(unresolved.len());
+        // now we need to resolve all imports for the source file and those imported from other
+        // locations
+        while let Some((path, node)) = unresolved.pop_front() {
+            let mut resolved_imports = Vec::with_capacity(node.data.imports.len());
+
+            // parent directory of the current file
+            let node_dir = path.parent();
+            if node_dir.is_none() {
+                continue
+            }
+            let node_dir = node_dir.unwrap();
+
+            for import in node.data.imports.iter() {
+                let component = import.components().next();
+                if component.is_none() {
+                    continue
+                }
+                let component = component.unwrap();
+                if component == Component::CurDir || component == Component::ParentDir {
+                    // if the import is relative we assume it's already part of the processed input
+                    // file set
+                    match utils::canonicalize(node_dir.join(import)) {
+                        Ok(target) => {
+                            // the file at least exists,
+                            if let Some(idx) = index.get(&target).cloned() {
+                                resolved_imports.push(idx);
+                            } else {
+                                // imported file is not part of the input files
+                                let node = read_node(&target)?;
+                                unresolved.push_back((target.clone(), node));
+                                let idx = index.len();
+                                index.insert(target.clone(), idx);
+                                resolved_imports.push(idx);
+                            }
+                        }
+                        Err(err) => {
+                            tracing::trace!("failed to resolve relative import \"{:?}\"", err);
```

Should we silently ignore this error or should we throw?

---

this can happen if the import path is wrong, points to a file that does not exist.
so the expectation here would be that this will fail in solc regardless but with a better error message.